### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by didva**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import './App.css';
+import {BrowserRouter, Route, Routes} from 'react-router-dom'
```
Comment (didva) [inline src/App.js:2]: Fix quotation

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import React, {Component} from "react";
+import Header from './components/layout/Header'
+import Footer from "./components/layout/Footer";
+import Catalogue from "./components/pages/Catalogue";
+import Volume from "./components/pages/Volume";
+import Search from "./components/pages/Search";
+
+class App extends Component {
```
Comment (didva) [inline src/App.js]: Function component

File: src/components/Filter.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+function Filter(props) {
```
Comment (didva) [inline src/components/Filter.js]: Arrow function

File: src/components/Filter.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        <div className="filter-container">
+            <div className="filter-name">{props.name}</div>
+            <div className="filter">
+                <input type="text" value={props.value} onChange={onChange} id={props.id}/>
+            </div>
+        </div>
+    );
+}
+
+export default Filter;
```
Comment (didva) [inline src/components/Filter.js:14]: Be consistent with export default.

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import './App.css';
```
Comment (didva) [inline src/App.js:1]: Don't use global css

File: src/components/SearchBox.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import {Link, useNavigate} from 'react-router-dom'
+import Thumbnail from './Thumbnail';
+
+function SearchBox(props) {
+    const [search, setSearch] = useState('');
+    const [results, setResults] = useState([]);
+    const url = props.baseUrl + '/volumes?key=' + props.apiKey;
+    const navigate = useNavigate();
+
+    useEffect(() => {
```
Comment (didva) [inline src/components/SearchBox.js:12]: Create service for API

File: src/components/Thumbnail.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+function Thumbnail(props) {
```
Comment (didva) [inline src/components/Thumbnail.js]: Use destructing everywhere

File: src/components/Filter.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+function Filter(props) {
+    const onChange = (e) => {
```
Comment (didva) [inline src/components/Filter.js]: useCallback

File: src/components/FiltersList.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import Filter from "./Filter";
+
+function FiltersList({filters, setFilters, onSubmit}) {
+
+    const onChange = (id, value) => {
```
Comment (didva) [inline src/components/FiltersList.js]: useCallback

File: src/components/Rating.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+export default function Rating({volumeInfo}) {
+    return (
+        <React.Fragment>
+            {volumeInfo.averageRating &&
+                <React.Fragment>
```
Comment (AndriiHopko) [inline src/components/Rating.js]: <>
   {asd}
</>

Comment (didva) [inline src/components/Rating.js]: <></> instead <React.Fragment></React.Fragment>

File: src/components/SearchBox.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+                    headers: {
+                        'Accept': 'application/json',
+                        'Content-Type': 'application/json'
+                    }
+                }).then(response => response.json())
+                    .then(results => setResults(() => results.items));
+            }, 1000);
+            return () => { clearTimeout(timeout) };
+        } else {
+            setResults(() => []);
```
Comment (AndriiHopko) [inline src/components/SearchBox.js]: setResults([]);

File: src/components/SearchBox.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        }
+    }, [url, search]);
+
+    const onSubmit = function (e) {
+        e.preventDefault();
+    };
+
+    const onChange = function (e) {
```
Comment (AndriiHopko) [inline src/components/SearchBox.js]: (e) => {}

File: src/components/Thumbnail.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    return (
+        <React.Fragment>
+            {props.volumeInfo.imageLinks && props.volumeInfo.imageLinks.thumbnail &&
```
Comment (AndriiHopko) [inline src/components/Thumbnail.js]: props.volumeInfo.imageLinks && props.volumeInfo.imageLinks.thumbnail => props.volumeInfo?.imageLinks?.thumbnail

File: src/utils/SearchUtils.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
+export function getFiltersObject(searchParams, filtersList) {
+    let filters = {};
+    for (const filter of filtersList) {
```
Comment (AndriiHopko) [inline src/utils/SearchUtils.js]: forEach([], ()=> {});
filtersList.reduce((acc, array) => {

}, {})

File: src/components/GAuth.js [hunk 15]
```diff
@@ -0,0 +1,66 @@
+import React, {useCallback, useContext, useEffect} from 'react';
+import { googleLogout, useGoogleLogin } from "@react-oauth/google";
+import UserContext from "../contexts/UserContext";
+
+const GAuth = () => {
+    const {user, setUser} = useContext(UserContext);
+    const onSuccess = useCallback((response) => {
+        const token = response.access_token;
+        const expiresIn = response.expires_in;
+        fetch(`https://www.googleapis.com/oauth2/v1/userinfo?access_token=${token}`, {
```
Comment (didva) [inline src/components/GAuth.js]: service

File: src/components/Loader.js [hunk 16]
```diff
@@ -1,11 +1,11 @@
...
+import React from 'react';
 import loader from '../static/loader.gif';
 
-function Loader(props) {
-    const [loading, setLoading] = useState(props.loading);
-    useEffect(() => {
-        setLoading(() => props.loading);
-    }, [props.loading]);
+const Loader = ({loading}) => {
+    // const [loading, setLoading] = useState(props.loading);
```
Comment (didva) [inline src/components/Loader.js]: remove comments

File: src/components/VolumeList.js [hunk 17]
```diff
@@ -3,50 +3,50 @@ import {Link} from 'react-router-dom'
...
+        });
+    }, [query, booksApiService, removedTrigger, queryPath]);
 
     return (
-        <React.Fragment>
+        <>
             <Loader loading={loading}/>
             <div>
-                {!loading && volumes && volumes.map(volume =>
+                {!loading && volumes?.length > 0 && volumes.map(volume =>
```
Comment (didva) [inline src/components/VolumeList.js]: ternary operator?

**Pull request merged**